announcement/announcement-impl/impl/src/java/org/sakaiproject/announcement/impl/AnnouncementObserver.java

```diff
@@ -0,0 +1,73 @@
+package org.sakaiproject.announcement.impl;
+
+import java.util.List;
+import java.util.Observable;
+import java.util.Observer;
+
+import org.sakaiproject.announcement.api.AnnouncementService;
+import org.sakaiproject.event.api.Event;
+import org.sakaiproject.event.api.EventTrackingService;
+import org.sakaiproject.memory.api.Cache;
+import org.sakaiproject.memory.api.MemoryService;
+import org.sakaiproject.message.api.Message;
+
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class AnnouncementObserver implements Observer {
+
+    @Setter
+    private EventTrackingService eventTrackingService;
+    @Setter
+    private MemoryService memoryService;
+    private Cache<String, List<Message>> messagesCache;
+
+    public void init() {
+        log.info("init()");
+        eventTrackingService.addLocalObserver(this);
+        messagesCache = memoryService.getCache("org.sakaiproject.announcement.tool.messages.cache");
+    }
+
+    public void destroy() {
+        log.info("destroy");
+        eventTrackingService.deleteObserver(this);
+    }
+
+    public void update(Observable arg0, Object arg) {
+        if (!(arg instanceof Event)) {
+            return;
+        }
+
+        final Event event = (Event) arg;
+        final String eventType = event.getEvent();
+
+        if (eventType != null) {

```

Comment:
since all the switch blocks call the same code:
is there really a need for any of the breaks?

I would refactor to something like:
```
if (eventType != null) {
  String channelName = getChannel(event);
  switch(eventType) {
    case AnnouncementService.SECURE_ANNC_ADD:
    case AnnouncementService.SECURE_ANNC_UPDATE_ANY:
    case AnnouncementService.SECURE_ANNC_UPDATE_OWN:
    case AnnouncementService.SECURE_ANNC_REMOVE_ANY:
    case AnnouncementService.SECURE_ANNC_REMOVE_OWN:
      log.debug("Announcement event: {}", eventType); 
      messagesCache.remove(channelName);
      break;
    default:
      break;
  }
}
```

Reply:
Good point! Thanks for the advice!